spec/connect/client_spec.rb

```diff
@@ -548,6 +613,18 @@
     end
   end
 
+  describe '#flatten_tree' do
+    it 'returns all products in a tree' do
+      result = subject.flatten_tree(product).uniq
+      expect(result.size).to eq 8
```

Comment:
Why `uniq`? Also, it would be better to do `.map(&:identifier)` and verify the values instead of the array size.